sys/net/application_layer/emcute/emcute_str.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_emcute
+ * @{
+ *
+ * @file
+ * @brief       emCute string functions (for debugging purposes)
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "net/emcute.h"
+#include "emcute_internal.h"
+
+const char *emcute_type_str(uint8_t type)
+{
+    switch (type) {
+        case ADVERTISE:     return "ADVERTISE";
```

Comment:
actually, gcc should convert this to a lookup table, as the type IDs are up-counting without gaps... Also its just for debugging, so efficiency is not uttermost important...